src/sage/geometry/hyperplane_arrangement/arrangement.py

```diff
@@ -758,8 +763,13 @@ def cone(self, variable='t'):
         hyperplanes.append([0, 1] + [0] * self.dimension())
         P = self.parent()
         names = (variable,) + P._names
-        H = HyperplaneArrangements(self.parent().base_ring(), names=names)
-        return H(*hyperplanes, backend=self._backend)
+        from sage.geometry.hyperplane_arrangement.ordered_arrangement import OrderedHyperplaneArrangements
+        if isinstance(P, OrderedHyperplaneArrangements):
+            H = OrderedHyperplaneArrangements(self.parent().base_ring(), names=names)
+        else:
+            H = HyperplaneArrangements(self.parent().base_ring(), names=names)
```

Comment:
```suggestion
        H = type(P).__base__(P.base_ring(), names=names)
```
A simplification (note that the type of `P` is mutated by the category construction, hence the `__base__` call) and future proofing.

Reply:
Changed, I took out also the previous line `from sage.geometry.hyperplane_arrangement.ordered_arrangement import OrderedHyperplaneArrangements`